src/Interfaces/AbstractZonotope.jl

```diff
@@ -525,3 +525,52 @@ function constraints_list(Z::AbstractZonotope{N}; check_full_rank::Bool=true
     end
     return constraints
 end
+
+"""
+    rectify(Z::AbstractZonotope{N}) where {N}
+
+Concrete rectification of a zonotopic set.
+
+### Input
+
+- `Z` -- zonotope
+
+### Output
+
+The zonotope obtained by rectifying `Z`.
+
+### Algorithm
+
+This function implements [Theroem 3.1, 1].
+
+[1] *Singh, G., Gehr, T., Mirman, M., Püschel, M., & Vechev, M. (2018). Fast
+and effective robustness certification. In Advances in Neural Information
+Processing Systems (pp. 10802-10813).*
+"""
+function rectify!(Z::AbstractZonotope{N}) where {N}
+    c = Z.center
+    G = Z.generators
+    n, m = size(G)
+    H = overapproximate(Z, Hyperrectangle)
+    Gnew = zeros(N, n, n)
+
+    @inbounds for i in 1:n
+        lx, ux = H.center[i] - H.radius[i], H.center[i] + H.radius[i]
+        if !_leq(lx, zero(N))
+            nothing
+        elseif _leq(ux, zero(N)) || isapproxzero(lx)
+            c[i] = zero(N)
+            G[i, :] = zeros(N, m)
+        else
+            λ = ux / (ux - lx)
+            μ = - λ * lx / 2
+            c[i] = c[i] * λ + μ
+            G[i, :] = G[i, :] .* λ
+            Gnew[i, i] = μ
+        end
+    end
+    Z = Zonotope(c, hcat(G, Gnew))
+    return remove_zero_generators(Z)
+end
+
+rectify(Z::AbstractZonotope) = rectify!(copy(Z))
```

Comment:
Better move these methods to the `Approximations` module because they are not exact.

Reply:
About the place of this function, in fact i suggest to place this in `Zonotope.jl`. An inconvenience that i find by dispatching on top level set interfaces like that is that there are little assumptions you can make on the type, and you end up hard-coding `Vector{N}` and so on which is in many cases something one *doesn't* wan't to do, because that may change the type of the input zonotope. 

For example, `split` or `reduce_order` zonotope functions are *much* faster if you use small statically sized arrays, and we have versions that specialize on those ([here](https://juliaarrays.github.io/StaticArrays.jl/stable/pages/api/#Indexing-1) and [here](https://github.com/JuliaReach/ReachabilityAnalysis.jl/blob/master/src/Flowpipes/setops.jl#L604) . Taking the vec rep and the mat rep from the dispatch on the abstract type is possible but cumbersome and anit-julian.

In sum, a possible pattern is to mimic the one we have in place the concrete 
linear map function `linear_map` as shown below.

```julia
# in file AbstractZonotope.jl
#  fallback
rectify(Z::AbstractZonotope) = _rectify(Zonotope(center(Z), genmat(Z))

# in file Zonotope.jl, use getter function
function _rectify(Z::Zonotope)
    # .... new method
end

# another _rectify for another zonotope type
function _rectify(Z::Zonotope)
    # .... new method
end
```